### DISCUSSION TITLE
How could i delete useless css when vite build

### DISCUSSION THREAD

**Discussion by wangyuanfen** [tailwindlabs/tailwindcss#14271] [Help]
**What version of Tailwind CSS are you using?**

v3.4.10

**What build tool (or framework if it abstracts the build tool) are you using?**

vite 5.4.1

**What version of Node.js are you using?**

v20.13.1

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

https://github.com/wangyuanfen/tailwind

**Describe your issue**

when i build , my App.tsx no import sub.tsx

App.tsx
```
export default function App() {
  return (
    <h1 className="text-3xl font-bold underline">
      Hello world!
    </h1>
  )
}
```

sub.tsx
```
export default function Test() {
  return (
    <h1 className="text-[#123456] bg-[url(./avatar.jpg)]">
      12345789
    </h1>
  )
}
```

and the dist have the useless css and picture
![1](https://github.com/user-attachments/assets/ff8888f3-fdb6-4bb3-9a58-a184a0e539b1)

Doesn't JIT delete useless CSS?

I hope the useless css not show in the dist, especially the jpg no build in dist

**Comment by wongjn** [comment]
You could exclude the `sub.tsx` file like:

```diff
 /** @type {import('tailwindcss').Config} */
 export default {
   content: [
     "./index.html",
     "./src/**/*.{js,ts,jsx,tsx}",
+    "!./src/sub.tsx",
  ],
```
Now, Tailwind won't scan the `sub.tsx` and won't generate any classes that appear inside `sub.tsx`, if they don't appear anywhere else in content that Tailwind *does* scan.

**Reply by wangyuanfen** [reply]
> You could exclude the `sub.tsx` file like:
> 
> ```diff
>  /** @type {import('tailwindcss').Config} */
>  export default {
>    content: [
>      "./index.html",
>      "./src/**/*.{js,ts,jsx,tsx}",
> +    "!./src/sub.tsx",
>   ],
> ```
> 
> Now, Tailwind won't scan the `sub.tsx` and won't generate any classes that appear inside `sub.tsx`, if they don't appear anywhere else in content that Tailwind _does_ scan.

Is there anything useless that I need to configure in Content? Rollup won’t shake?

**Reply by wongjn** [reply]
Rollup has no participation in the CSS that Tailwind generates.

**Reply by wangyuanfen** [reply]
> Rollup has no participation in the CSS that Tailwind generates.

So, if i have some common file no useless,  i must to exclude in content, even thought i never use in any other file, cause tailwindcss scanner all file ?

**Reply by wongjn** [reply]
Indeed.
